File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/monsters/LevelUpReport.java
package de.limbusdev.guardianmonsters.guardians.monsters;

/**
 * LevelUpReport
 *
 * @author Georg Eckert 2017
 */

public class LevelUpReport
{
    public Statistics oldStats;
    public Statistics newStats;
    public int oldLevel, newLevel;

    public LevelUpReport(Statistics oldStats, Statistics newStats, int oldLevel, int newLevel)
    {
        this.oldStats = oldStats;
        this.newStats = newStats;
        this.oldLevel = oldLevel;
        this.newLevel = newLevel;
    }

    /**
     * For Serialization only!
     */
    LevelUpReport() {}
}

File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/monsters/LevelUpReport.kt
package de.limbusdev.guardianmonsters.guardians.monsters

/**
 * LevelUpReport
 *
 * @author Georg Eckert 2017
 */

class LevelUpReport
(
        var oldStats: Statistics = Statistics(),
        var newStats: Statistics = Statistics(),
        var oldLevel: Int = 0,
        var newLevel: Int = 0

) {
    override fun toString(): String
    {
        return  "Before: Lvl $oldLevel, $oldStats\n" +
                "After:  Lvl $newLevel, $newStats"
    }
}
